src/main/java/com/google/sps/app/servlets/NewTripServlet.java

```diff
@@ -73,8 +72,8 @@ public String validateInput(HttpServletRequest request) {
         } catch (NumberFormatException e) {
             return "Invalid totalBudget";
         }
-        Date start = parseInputDate(request.getParameter(START_DATE_PARAM));
-        Date end = parseInputDate(request.getParameter(END_DATE_PARAM));
+        Date start = dataStoreHelper.parseInputDate(request.getParameter(START_DATE_PARAM));
```

Comment:
To avoid getting the dates twice from the cloud, pass them into the validator as an argument.